packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -518,29 +492,117 @@ function runInstructionSync(
-  }
-  if(instruction.skip) {
-    //ignore the result, SKIP/PAGE is not relevant for us, just for side-effects
-    evaluateExpression(instruction.skip, context);
+    const text = valueToString(expression) ?? "";
+    let tokens = lex(text);
+    if (instruction.scanMode !== inst.ScanMode.NoScan) {
+      tokens = replaceTokensInText(tokens, context);
+    }
+    if (breakCount > 0) {
+      context.tokens.push(...tokens);
+    } else {
+      if (
+        context.tokens.length > 0 &&
+        tokenOfType(
+          context.tokens[context.tokens.length - 1],
+          PreprocessorTokens.Id,
+        ) &&
+        tokens.length > 0 &&
+        tokenOfType(tokens[0], PreprocessorTokens.Id)
+      ) {
+        const lastToken = context.tokens.pop()!;
+        const firstToken = tokens.shift()!;
+        const newToken: Token = mergeIdTokens(lastToken, firstToken);
+        context.tokens.push(newToken);
+      }
+      context.tokens.push(...tokens);
+    }
   }
-  if(instruction.column) {
-    //ignore the result, COLUMN is not relevant for us, just for side-effects
-    evaluateExpression(instruction.column, context);
+  if (instruction.column) {
+    const columnCount = evaluateExpression(instruction.column, context);
+    if (typeof valueToNumber(columnCount) !== "number") {
+      context.diagnostics.push(
+        diagnosticFromCode(
+          Severe.IBM3948I,
+          instruction.columnToken,
+          "CONVERSION",
+          "612",
+        ),
+      );
+    }
   }
-  if(instruction.margins) {
-    //ignore the result, MARGINS is not relevant for us, just for side-effects
-    instruction.margins.left && evaluateExpression(instruction.margins.left, context);
-    instruction.margins.right && evaluateExpression(instruction.margins.right, context);
+  if (instruction.margins) {
+    let hasError = false;
+    if (instruction.margins.left) {
+      const leftCount = evaluateExpression(instruction.margins.left, context);
+      if (typeof valueToNumber(leftCount) !== "number") {
+        hasError = true;
+      }
+    }
+    if (instruction.margins.right) {
+      const rightCount = evaluateExpression(instruction.margins.right, context);
+      if (typeof valueToNumber(rightCount) !== "number") {
+        hasError = true;
+      }
+    }
+    if (hasError) {
+      context.diagnostics.push(
+        diagnosticFromCode(
+          Severe.IBM3948I,
+          instruction.marginsToken,
+          "CONVERSION",
+          "612",
+        ),
+      );
+    }
   }
 }
 
+function mergeIdTokens(leftId: Token, rightId: Token): Token {
+  return {
+    image: leftId.image + rightId.image,
+    originalImage: leftId.originalImage + rightId.originalImage,
+    startOffset: leftId.startOffset,
+    endOffset: rightId.endOffset,
+    tokenType: PreprocessorTokens.Id,
+    element: undefined,
+    kind: undefined,
+    immediateFollow: false,
+    tokenTypeIdx: leftId.tokenTypeIdx,
+    uri: undefined,
+    isInsertedInRecovery: false,
+  };
+}
```

Comment:
Suggestion: I don't think this logic is correct. Which is why `runTokenInstruction` (now `mergePush`) is using `lex` to generate the merged token. Something like `DE` and `CLARE` are individually just IDs, but when they're merged they become `DECLARE`, which is a different token type.

Reply:
Done.